arbnode/espresso_caff_node.go

```diff
@@ -350,29 +434,29 @@ func (n *EspressoCaffNode) createBlock() (returnValue bool) {
 	blockCalcTime := time.Since(startTime)
 
 	log.Info("Produced block", "block", block.Hash(), "blockNumber", block.Number(), "receipts", len(receipts))
-
-	err = n.executionEngine.AppendBlock(block, statedb, receipts, blockCalcTime)
+  
+  err = n.espressoStreamer.StoreHotshotBlock(n.db, messageWithMetadataAndPos.HotshotHeight)
 	if err != nil {
-		log.Error("Failed to append block", "err", err)
+		log.Error("Failed to store hotshot block", "err", err)
 		log.Debug("Resetting espresso streamer", "currentMessagePos",
 			messageWithMetadataAndPos.Pos, "currentHostshotBlock",
 			messageWithMetadataAndPos.HotshotHeight)
 		n.espressoStreamer.Reset(messageWithMetadataAndPos.Pos, messageWithMetadataAndPos.HotshotHeight)
 		return false
 	}
 
-	n.espressoStreamer.RecordTimeDurationBetweenHotshotAndCurrentBlock(messageWithMetadataAndPos.HotshotHeight, time.Now())
-
-	err = n.espressoStreamer.StoreHotshotBlock(n.db, messageWithMetadataAndPos.HotshotHeight)
+	err = n.executionEngine.AppendBlock(block, statedb, receipts, blockCalcTime)
 	if err != nil {
-		log.Error("Failed to store hotshot block", "err", err)
+		log.Error("Failed to append block", "err", err)
```

Comment:
I don't think we should store the hotshot block number before appending it.

The caff node will read the next hotshot block when it restarts. If the previous block isn't successfully appended, the caff node will skip this block. On the other hand, it's ok for us create a same block because it won't change the l2 state and this error will be skipped

Reply:
Is it not an issue for non restarting nodes? If we reset fail to add the message to the db after we add the block to the chain and reset to the message position that was just added to the chain as? Lets say that happens for message position n. Wouldn't that cause the next call to create block to grab another message at position n and fail to execute it or add it to the chain, which would result in create block calling reset on position n again? This would require a manual intervention to restart the node and get it unstuck.

Reply:
I see. I think maybe we should reorg the L2 chain data when reset is triggered.